ci use ` --force-with-lease`  when pushing to remote branch

alan-agius4 opened the pull request:
Tentative fix for:
```
! [rejected]        20.0.x -> 20.0.x (fetch first)
error: failed to push some refs to 'https://github.com/angular-robot/angular.git'
```

josephperrott commented on .github/workflows/sync-angular-robot-forked-repo.yml:
```diff
@@ -26,6 +26,7 @@ jobs:
         uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
         with:
           ref: ${{ github.ref }} # Checks out the branch that triggered the push
+          fetch-depth: 0 # Full clone history
```
Rather than the full clone history (which is ever growing and takes a while on here at times), should we just do something like 250 since that should be way more than we should need?

alan-agius4 replied:
Change it to 100 as that should suffice.

josephperrott reviewed (approved):
LGTM

alan-agius4 committed:
ci: use ` --force-with-lease`  when pushing to remote branch

Tentative fix for:
```
! [rejected]        20.0.x -> 20.0.x (fetch first)
error: failed to push some refs to 'https://github.com/angular-robot/angular.git'
```

alxhub commented:
This PR was merged into the repository by commit adffb2ec8257767c2149a5d6ad59894b4d968b09.

The changes were merged into the following branches: main, 19.2.x, 20.0.x
